sdk/core/switcher/entry.S

```diff
@@ -183,159 +188,266 @@ switcher_scheduler_entry_csp:
 	.p2align 2
 	.type __Z26compartment_switcher_entryz,@function
 __Z26compartment_switcher_entryz:
+	/*
+	 * Atlas:
+	 *  mtdc: thread trusted stack pointer
+	 *  ra: caller return address (nominally; corruption is on them)
```

Comment:
> (nominally; corruption is on them)

I think that this is entered via an interrupt disabling sentry so we can be sure that `ra` is a return sentry to the caller. If we ever change that to interrupt inheriting we'd lose that guarantee but I agree it doesn't matter much.

Reply:
That's a good point; will update.